sql/handler.h

```diff
@@ -4030,6 +4030,10 @@ class handler :public Sql_alloc
   virtual int ft_read(uchar *buf) { return HA_ERR_WRONG_COMMAND; }
   virtual int rnd_next(uchar *buf)=0;
   virtual int rnd_pos(uchar * buf, uchar *pos)=0;
+
+  virtual int sample_next(uchar *buf) { return 0;}
```

Comment:
I would suggest an interface that tells the storage engine beforehand how many rows (percentage wise or number wise) will be sampled. That opens up a number of optimizations, such as allowing the storage engine to pre-fetch those rows before sample_next takes place.

Reply:
It can be a separate call, like a condition pushdown (as an index_* alternative I guess). 
Anyway, using it is out of this task's scope, so this parameter will hang dangling, until somebody implements it, or not.